io-native/src/main/java/org/ballerinalang/stdlib/io/nativeimpl/CharacterChannelUtils.java

```diff
@@ -139,6 +184,21 @@ public static Object close(BObject channel) {
         return null;
     }
 
+    public static Object closeBufferedReader(BObject channel) {
+        try {
+            if(channel.getNativeData(BUFFERED_READER_ENTRY) != null) {
+                BufferedReader bufferedReader = (BufferedReader)
+                        channel.getNativeData(BUFFERED_READER_ENTRY);
+                bufferedReader.close();
+            }
+        } catch (ClosedChannelException e) {
+            return IOUtils.createError("channel already closed.");
+        } catch (IOException e) {
+            return IOUtils.createError(e);
+        }
```

Comment:
Can these be util functions?, Seems repeating for every channel?

Reply:
Fixed in here https://github.com/ballerina-platform/module-ballerina-io/pull/31/commits/98acb67cdbfd15b787de2bc02c14a106f0615bfe

Reply:
I meant the entire closefunction can be one common util class including try and catch as well, which can be called from respective readers

Reply:
Here(https://github.com/ballerina-platform/module-ballerina-io/commit/98acb67cdbfd15b787de2bc02c14a106f0615bfe) the repetition of BufferedReader null check has removed since it's unnecessary. Only BufferedInputStream null check is there and it won't repeat in multiple classes. But as you said, BufferedInputStream null check will repeat in multiple APIs, so I added an internal function to remove that repetition.

Reply:
Ack